crates/ruff_python_formatter/src/statement/stmt_delete.rs

```diff
@@ -1,12 +1,46 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
-use ruff_formatter::{write, Buffer, FormatResult};
-use rustpython_parser::ast::StmtDelete;
+use crate::prelude::PyFormatContext;
+use crate::{expression::parentheses::Parenthesize, AsFormat, FormatNodeRule, PyFormatter};
+use ruff_formatter::prelude::{format_with, space, text, Formatter};
+use ruff_formatter::{write, Buffer, Format, FormatResult};
+use rustpython_parser::ast::{Expr, StmtDelete};
 
 #[derive(Default)]
 pub struct FormatStmtDelete;
 
 impl FormatNodeRule<StmtDelete> for FormatStmtDelete {
     fn fmt_fields(&self, item: &StmtDelete, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let StmtDelete { range: _, targets } = item;
+        write!(f, [text("del"), space(), DeleteList::new(targets)])
+    }
+}
+
+// TODO(cnpryer): Impl FormatRuleWithOptions Parenthesize
+#[derive(Debug)]
+struct DeleteList<'a> {
+    delete_list: &'a [Expr],
+}
+
+impl<'a> DeleteList<'a> {
+    const fn new(delete_list: &'a [Expr]) -> Self {
+        Self { delete_list }
+    }
+}
+
+impl Format<PyFormatContext<'_>> for DeleteList<'_> {
+    fn fmt(&self, f: &mut Formatter<PyFormatContext<'_>>) -> FormatResult<()> {
+        write!(
+            f,
+            [&format_with(|f| {
+                let separator = text(", "); // TODO(cnpryer)
```

Comment:
i think you're looking for `soft_line_break_or_space()`

Reply:
a98d7213c1e1c241ff9aeceb133c0552a8b19060